api/dataprocessor.go

```diff
@@ -166,22 +147,22 @@ func aggEvery(numPoints, maxPoints uint32) uint32 {
 }
 
 // error is the error of the first failing target request
-func getTargets(store mdata.Store, reqs []Req) ([]Series, error) {
-	seriesChan := make(chan Series, len(reqs))
+func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
+	seriesChan := make(chan models.Series, len(reqs))
 	errorsChan := make(chan error, len(reqs))
 	// TODO: abort pending requests on error, maybe use context, maybe timeouts too
 	wg := sync.WaitGroup{}
 	wg.Add(len(reqs))
 	for _, req := range reqs {
-		go func(wg *sync.WaitGroup, req Req) {
+		go func(wg *sync.WaitGroup, req models.Req) {
 			pre := time.Now()
-			points, interval, err := getTarget(store, req)
+			points, interval, err := s.getTarget(req)
 			if err != nil {
 				errorsChan <- err
```

Comment:
by taking away from doRecover() from getTarget, getTarget will never return an error. So:
* I think we should change getTarget signature to not return an error, and remove this errorsChan stuff
* I'm not sure what will happen to the other running getTarget() calls when the panic bubbles up. I think they will just keep running, executing needless work. Whatever happens, I think the behavior is the same before and after this code change. but we should probably do something smarter using ErrGroups where if one getTarget errors, we can abort all the other ones, in particular, cancel any pending CRR's.